pytorch_lightning/trainer/trainer.py

```diff
@@ -490,6 +490,38 @@ def _setup_on_init(self, num_sanity_val_steps: int) -> None:
 
         self.num_predict_batches = []
 
+    def _call_and_handle_interrupt(self, trainer_fn: Callable, *args: Any, **kwargs: Any):
+        r"""
+        Error handling, intended to be used only for main trainer function entry points (fit, validate, test, predict)
+        as all errors should funnel through them
+
+        Args:
+            trainer_fn: one of (fit, validate, test, predict)
+
+            *args/**kwargs: args to be passed to trainer_fn
+        """
+        try:
+            return trainer_fn(*args, **kwargs)
+        except KeyboardInterrupt:
+            rank_zero_warn("Detected KeyboardInterrupt, attempting graceful shutdown...")
+            # user could press Ctrl+c many times... only shutdown once
+            if not self.interrupted:
+                self.state.status = TrainerStatus.INTERRUPTED
+                self.on_keyboard_interrupt()
+                # same treatment as below
+                self.accelerator.teardown()
```

Comment:
it might be fine here though, but we should comment why this is the case if the handling is different

Reply:
is there any downside to just removing this as well?

Reply:
@SeanNaren @awaelchli what do you think?

Reply:
I don't understand why you are asking this question. teardown was never there in the first place (check the old code). 
Are you asking if we can remove `accelerator.on_train_end()`? 

As I said in another comment somewhere, this is currently only implemented by the IPU Plugin and if we ask @SeanNaren very nicely maybe he can check if keyboard interrupt works on the IPU or not (he has access to the IPU machine). I will ask him.

Reply:
thanks @awaelchli I think there was some confusion on my part here, sorry about that. it seems like we shouldn't have `self.accelerator.teardown()` here and I will remove it. It probably isnt necessary to check on IPU, but feel free